paddle/framework/executor.cc

```diff
@@ -116,6 +117,11 @@ void Executor::Run(const ProgramDesc& pdesc, Scope* scope, int block_id,
   for (auto& op_desc : block.AllOps()) {
     auto op = paddle::framework::OpRegistry::CreateOp(*op_desc);
     VLOG(3) << op->DebugStringEx(local_scope);
+
+    platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
+    auto dev_ctx = const_cast<platform::DeviceContext*>(pool.Get(place_));
+    platform::RecordEvent record_event(op->Type(), dev_ctx);
```

Comment:
`RecordEvent` should not be always called,  only when `ProfilerState` is not `kDisabled`.

Reply:
@chengduoZH  Thanks for your review. Whether to record timeline is judged in the constructor of `RecordEvent`: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/platform/profiler.cc#L130